replication: don't rollback qsync limbo wait on fiber cancel (#9465)
tarantool/tarantool: changelogs/unreleased/gh-9480-dont-rollback-on-quorum-wait-cancel.md

## bugfix/replication

> * Fixed to not rollback a transaction waiting for quorum from replicas
>   on transaction fiber cancel (gh-9480).

Reviewer: ```suggestion
* Now transactions are not rolled back if the transaction fiber is
  cancelled when waiting for quorum from replicas (gh-9480).
```

Author: Done.
